File: app/src/main/java/com/my/televip/virtuals/ActionBar/Theme.java
package com.my.televip.virtuals.ActionBar;

import android.text.TextPaint;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class Theme {

    public static TextPaint getTextPaint()
    {
        return (TextPaint) XposedHelpers.getStaticObjectField(ClassLoad.getClass(ClassNames.THEME), Obfuscate.getFieldName("Theme", "chat_timePaint"));
    }

    public static boolean isLight(){
        return !((boolean)XposedHelpers.callStaticMethod(ClassLoad.getClass(ClassNames.THEME), Obfuscate.getMethodName("Theme", "isCurrentThemeDark")));
    }
}

File: app/src/main/java/com/my/televip/virtuals/ActionBar/Theme.kt
package com.my.televip.virtuals.ActionBar

import android.text.TextPaint
import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

object Theme {

    @JvmStatic
    fun getTextPaint(): TextPaint? {
        val themeClass = ClassLoad.getClass(ClassNames.THEME) ?: return null
        return XposedHelpers.getStaticObjectField(
            themeClass,
            Obfuscate.getFieldName("Theme", "chat_timePaint")
        ) as? TextPaint
    }

    @JvmStatic
    fun isLight(): Boolean {
        val themeClass = ClassLoad.getClass(ClassNames.THEME) ?: return true
        val isDark = XposedHelpers.callStaticMethod(
            themeClass,
            Obfuscate.getMethodName("Theme", "isCurrentThemeDark")
        ) as? Boolean ?: false
        return !isDark
    }
}
